File: src/view/HistoryView.java
package view;

import model.History;

import java.util.List;

public class HistoryView {
    public static void printHistory(List<History> historyList) {
        BarView.bar("History");
        if (historyList.isEmpty()) {
            System.out.println("Empty history");
            return;
        }
        for (int i = 0; i < historyList.size(); i++) {
            History history = historyList.get(i);
            System.out.println((i + 1) + ". " + history.getTimestamp().toString() + " - " + history.getOperation() + " - " + "sold: " + history.getWallet().getSold());
        }
    }
}

File: src/view/history.view.ts
import { OperationHistory } from "../model/operation.history.model";
import { bar } from "./bar.view";

export const printHistory = (historyList: OperationHistory[]): void => {
  bar("History");
  if (historyList.length == 0) {
    console.log("Empty history");
    return;
  }
  for (let i = 0; i < historyList.length; i++) {
    console.log(
      `${i + 1}. ${historyList[i].getTimestamp().toLocaleDateString("fr")} - Operation: ${historyList[i].getOperation()} - sold: ${historyList[i].getWallet().getSold()}`
    );
  }
};
